pkg/detectors/sendgrid/sendgrid.go

```diff
@@ -67,16 +74,18 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 				// 403 means good key but not the right scope
 				// 401 is bad key
 				if res.StatusCode == http.StatusOK || res.StatusCode == http.StatusForbidden {
-					s.Verified = true
+					s1.Verified = true
```

Comment:
question: How come we are setting verified to true if the response has a 403 status code. Is it b/c we are assuming the credential is valid, and it just doesn't have access to this particular resource? Are we also certain that the `/templates` endpoint is the least restrictive, while still requiring auth? This might also just also just be a normal pattern with detectors where we consider 403s verified.

Reply:
this 403 being valid but incorrect scope might be a sendgrid specific thing. i've gotten 403 on completely invalid API keys before.

Reply:
> How come we are setting verified to true if the response has a 403 status code. Is it b/c we are assuming the credential is valid, and it just doesn't have access to this particular resource? 
correct
> Are we also certain that the /templates endpoint is the least restrictive, while still requiring auth? 
no not certain, it's what we had before - there might potentially be a better endpoint but i didn't put in the time to research that  😅